src/renderers/dom/shared/syntheticEvents/SyntheticUIEvent.js

```diff
@@ -29,13 +29,7 @@ var UIEventInterface = {
       return target;
     }
 
-    var doc = target.ownerDocument;
-    // TODO: Figure out why `ownerDocument` is sometimes undefined in IE8.
-    if (doc) {
-      return doc.defaultView || doc.parentWindow;
-    } else {
-      return window;
-    }
+    return target.ownerDocument.defaultView;
```

Comment:
I believe you could still have cases where ownerDocument is undefined, e.g. if the node is detached (for some reason)

Reply:
This whole block might be able to go away, it looks like IE9 supports event.view:

<img width="503" alt="screen shot 2017-10-11 at 7 24 17 am" src="https://user-images.githubusercontent.com/590904/31437984-4304a2b0-ae55-11e7-8cbd-26a1973a48cf.png">

I'll do some testing. MDN isn't comprehensive on support here:

<img width="721" alt="screen shot 2017-10-11 at 7 24 57 am" src="https://user-images.githubusercontent.com/590904/31438003-549b0640-ae55-11e7-83cd-348e2c7e1d05.png">

Maybe we can fix that :)

Reply:
Yep! `UIEvent.view` is supported in:

IE9-11
Safari Desktop 4, 5.1, 6.2, 7.1, 10.1, 11.0
Safari iOS 3, 4
FF 47
Windows Phone 8.1

Here's my test case:
https://codepen.io/nhunzaker/pen/BwVwqo

And I've submitted a revision to this page on MDN:
https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/view